qutebrowser/misc/editor.py

```diff
@@ -68,29 +68,26 @@ def on_proc_closed(self, exitcode, exitstatus):
         Callback for QProcess when the editor was closed.
         """
         log.procs.debug("Editor closed")
-        if exitstatus != QProcess.NormalExit:
-            # No error/cleanup here, since we already handle this in
-            # on_proc_error.
+        if exitstatus != QProcess.NormalExit or exitcode != 0:
+            message.error("Editor processed failed to exit cleanly."
+                          "The file is still available at {}".format(
+                              self._file.name))
             return
+
+        encoding = config.get('general', 'editor-encoding')
         try:
-            if exitcode != 0:
-                return
-            encoding = config.get('general', 'editor-encoding')
-            try:
-                with open(self._file.name, 'r', encoding=encoding) as f:
-                    text = f.read()
-            except OSError as e:
-                # NOTE: Do not replace this with "raise CommandError" as it's
-                # executed async.
-                message.error("Failed to read back edited file: {}".format(e))
-                return
-            log.procs.debug("Read back: {}".format(text))
-            self.editing_finished.emit(text)
-        finally:
-            self._cleanup()
-
-    @pyqtSlot(QProcess.ProcessError)
-    def on_proc_error(self, _err):
+            with open(self._file.name, 'r', encoding=encoding) as f:
+                text = f.read()
+        except (OSError, UnicodeError, UnicodeDecodeError) as e:
+            # NOTE: Do not replace this with "raise CommandError" as it's
+            # executed async.
+            message.error("Failed to read back edited file: {}\n"
```

Comment:
I'm not sure I agree that's a problem - you can increase `ui -> message-timeout` or use `:messages` - and newer messages don't hide older ones anymore either.